indexer/src/fastnear_indexer.rs

```diff
@@ -0,0 +1,207 @@
+use std::collections::HashMap;
+use near_indexer::near_primitives::{types::AccountId, views::{ActionView, ExecutionStatusView, ReceiptEnumView}};
+use reqwest::Client;
+use tokio::sync::{mpsc::{Sender, Receiver}, mpsc};
+use tracing::{info, error, debug, trace};
+
+use crate::{errors::Error, rmq_publisher::{get_routing_key, PublishData, PublishOptions, PublishPayload, PublisherContext}, types::{BlockWithTxHashes, IndexerExecutionOutcomeWithReceiptAndTxHash, PartialCandidateData, PartialCandidateDataWithBlockTxHash}};
+use crate::metrics::{make_block_listener_metrics, BlockEventListener, Metricable};
+use crate::errors::Result;
+
+const FASTNEAR_ENDPOINT: &str = "https://testnet.neardata.xyz/v0/last_block/final";
+
+pub struct FastNearIndexer {
+    client: Client,
+    addresses_to_rollup_ids: HashMap<AccountId, u32>,
+    listener: Option<BlockEventListener>,
+}
+
+impl FastNearIndexer {
+    pub(crate) fn new(addresses_to_rollup_ids: HashMap<AccountId, u32>) -> Self {
+        debug!(target: "fastnear_indexer", "Creating new FastNearIndexer");
+        Self { 
+            client: Client::new(),
+            addresses_to_rollup_ids,
+            listener: None,
+        }
+    }
+
+    pub fn run(&self) -> Receiver<PublishData> {
+        info!(target: "fastnear_indexer", "Starting FastNearIndexer");
+        let block_receiver = self.stream_latest_blocks();
+        let (publish_sender, publish_receiver) = mpsc::channel(100);
+
+        let addresses_to_rollup_ids = self.addresses_to_rollup_ids.clone();
+
+        tokio::spawn(async move {
+            Self::process_blocks(block_receiver, publish_sender, addresses_to_rollup_ids).await;
+        });
+
+        publish_receiver
+    }
+
+    async fn process_blocks(
+        mut block_receiver: Receiver<BlockWithTxHashes>,
+        publish_sender: Sender<PublishData>,
+        addresses_to_rollup_ids: HashMap<AccountId, u32>,
+    ) {
+        debug!(target: "fastnear_indexer", "Starting block processing");
+        while let Some(block) = block_receiver.recv().await {
+            trace!(target: "fastnear_indexer", "Received block: {:?}", block.block.header.height);
+            if let Err(e) = Self::parse_and_publish_block(block, &publish_sender, &addresses_to_rollup_ids).await {
+                error!(target: "fastnear_indexer", "Error parsing and publishing block: {:?}", e);
+            }
+        }
+    }
+
+    async fn parse_and_publish_block(
```

Comment:
https://github.com/coderabbitai generate unit or integration testing code for this function